### PR TITLE
api keys 

### PR DESCRIPTION
**Pull request by melindabernrdo**
- [x] General API code cleanup  (comments/naming conventions)
- [x] Limit API keys to 3 active keys 
- [x] Base API key deletion on userAPKeyID 
- [x] Remove unused `setSessionExpirationTime`

### CODE REVIEW

File: accounts/api_keys.go [hunk 1]
```diff
@@ -2,44 +2,62 @@ package accounts
...
-	// Soft delete API key
-	err := a.databaseClient.DeleteAPIKey(userID, keyID)
+	keyExists, err := a.databaseClient.DeleteAPIKey(userID, keyID)
 	if err != nil {
-		if err.Error() == "Not found" {
-			return ErrTokenNotFound
-		}
 		return errors.Wrap(err, "DeactivateAPIKey: Error deactivating API key")
 	}
+	if !keyExists {
```
Comment (c9rl) [inline accounts/api_keys.go:17]: newline

File: accounts/api_keys.go [hunk 2]
```diff
@@ -2,44 +2,62 @@ package accounts
...
+func (a *AccountClient) CreateAPIKey(userID int) (*models.APIKey, error) {
+	apiKeysList, err := a.databaseClient.GetAPIKeys(userID)
+	if err != nil {
+		return nil, errors.Wrap(err, "CreateAPIKeys: Error getting list of API keys")
+	}
+
+	var activeKeysCount int
+	for _, key := range apiKeysList {
+		// api key is active (hasn't been deleted)
+		if !key.DeletedAt.Valid {
```
Comment (c9rl) [inline accounts/api_keys.go:34]: `key.DeletedAt.Valid` is true if the DeletedAt col is filled, correct?

File: accounts/api_keys.go [hunk 3]
```diff
@@ -2,44 +2,62 @@ package accounts
...
-		return APIKey, errors.Wrap(err, "CreateAPIKey: Error creating API key")
+		return nil, errors.Wrap(err, "CreateAPIKey: Error creating API key")
 	}
 
-	return APIKey, nil
+	return apiKey, nil
 }
 
 // ListAPIKeys lists all API keys associated with a user
 func (a *AccountClient) ListAPIKeys(userID int) (models.APIKeySlice, error) {
```
Comment (c9rl) [inline accounts/api_keys.go:56]: Is this used anywhere?

**Review by c9rl — APPROVED** [review]
Looks good to me. ⚡️

File: accounts/api_keys.go (hunk 1 above)
Comment (melindabernrdo) [reply accounts/api_keys.go:17]: I usually don't leave a space between the error and checking for nil statements :) I think this is a fairly common convention in Go.

File: accounts/api_keys.go (hunk 3 above)
Comment (melindabernrdo) [reply accounts/api_keys.go:56]: Are you referring to the `ListAPIKeys` method? If so, it's used by the GET route for API Keys.

**Pull request merged**